torch/onnx/_internal/fx/analysis/unsupported_nodes.py

```diff
@@ -61,14 +63,23 @@ def analyze(
                         node, self.diagnostic_context
                     )
                 except diagnostics.RuntimeErrorWithDiagnostic as e:
-                    unsupported_nodes.append(node)
-
-        op_to_target_mapping: Dict[str, Dict[str, None]] = {}
-
-        for node in unsupported_nodes:
-            op = node.op
-            target = node.target
-            op_to_target_mapping.setdefault(op, {}).setdefault(str(target), None)
+                    # Extract the dtype information from the diagnostic message.
+                    if e.diagnostic.message and e.diagnostic.message.startswith(
```

Comment:
can we raise specific exception type instead of string matching to detect this error? string matching on logs is usually flaky. we log in one place and check the string matching in another. Unless we have a test for guarding this string matching, it will eventually break

Reply:
We all know that. The root cause is that the mismatch message in SARIF and screen error message. I wonder if we should create a new rule to get a new template in this error message. @BowenBao Do you have any suggestion.

Reply:
Added test for the current approach

Reply:
Maybe we can just expand the error being emitted from this analysis, showing `diagnostic.message` for each op. That way we don't need to do string parsing on them.

Reply:
Or maybe we can make it standard to attach dtype with optype, through looking at the graph instead of string parsing.

Reply:
I see. The first one seems good and we can align the message in screen printing with SARIF.

Reply:
> Maybe we can just expand the error being emitted from this analysis, showing `diagnostic.message` for each op. That way we don't need to do string parsing on them.

Not relying on string matching is a nice improvement. Could we go further and populate `op_to_target_mapping` (thus the error message) without using diagnostic at all?

The diagnostics is a passive component that captures extra debug information and put on a SARIF file - only that. Ideally, it should not influence (by populating `op_to_target_mapping`) and alter `UnsupportedFxNodesAnalysisResult` outcome. 

Maybe I am missing something, but by reusing diagnostics for anything other than SARIF, we are creating a direct coupling between `UnsupportedFxNodesAnalysis` and SARIF; That is, `UnsupportedFxNodesAnalysis` cannot properly report the error without depending on diagnostics machinery.

IMO, if we could find a way to capture the missing information on the analysis pass independently of diagnostics, we can keep the analysis pass and diagnostics decoupled.

Reply:
Made some modifications. Eseentially, now we don't leverage on `get_function_overloads`, instead, we use `is_registered_op` to capture the ATen that doesn't even have at least one ONNX invariant. Completely extract out the arg/function sig matching from UnsupoortedNodeAnalysis.

If ATen is missing complex support, the error will be emitted during dispatcher now, which provides a more suitable error message.